src/mvba/mvba/error.rs

```diff
@@ -0,0 +1,22 @@
+use crate::mvba::{proposal::Proposal, NodeId};
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq)]
```

Comment:
You mentioned you wanted the `PartialEq` impl here for tests but as you can see, that will make using the `#[from]` macro impossible since Errors imported from other crates usually do not impl PartialEq / Eq.

What you should do instead of equality tests is to use match tests. I.e. in your tests: 

```rust
assert!(matches!(err, Error::Encoding(_)))
```

Reply:
I will try to remove `PartialEq`. Actually it is in TODO list. I don't like to modify code because of tests.